### PR TITLE
Feature : Adding percentage option to Input Number

### PR DESCRIPTION
**Pull request by guillim**
fixing #7375

### CODE REVIEW

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts [hunk 1]
```diff
@@ -410,7 +410,7 @@ export class FieldMetadataService extends TypeOrmQueryService<FieldMetadataEntit
       // We're running field update under a transaction, so we can rollback if migration fails
       await fieldMetadataRepository.update(id, fieldMetadataForUpdate);
 
-      const updatedFieldMetadata = await fieldMetadataRepository.findOne({
+      const [updatedFieldMetadata] = await fieldMetadataRepository.find({
```
Comment (Weiko) [inline packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts:413]: Why this change?

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata-validation.service.ts [hunk 2]
```diff
@@ -31,18 +33,22 @@ export class FieldMetadataValidationService<
...
-      if (
-        decimals !== undefined &&
-        (decimals < 0 || !Number.isInteger(decimals))
-      ) {
-        throw new FieldMetadataException(
-          `Decimals value "${decimals}" must be a positive integer`,
-          FieldMetadataExceptionCode.INVALID_FIELD_INPUT,
-        );
-      }
+    const schema = z.object({
```
Comment (Weiko) [inline packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata-validation.service.ts:36]: nit: I prefer to avoid zod in the backend and use class-validator instead since backend is more class oriented. It seems the package is already there so it's fine but I'll add to my TODO so we don't have both libs of validation and class-validator makes more sense in general so we will keep that one!

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts (hunk 1 above)
Comment (guillim) [reply packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts:413]: Totally decorelated from this PR. I should have made a side PR for this one. 

Reason : while pair prog, I remember @charlesBochet said `find` only cost 1 request instead of 2 with `findOne`

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata-validation.service.ts (hunk 2 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata-validation.service.ts:36]: Try with this if you want
```typescript
 enum ValueType {
  PERCENTAGE = 'percentage',
  NUMBER = 'number'
}

class SettingsValidation {
  @IsOptional()
  @IsInt()
  @Min(0)
  decimals?: number;

  @IsOptional()
  @IsEnum(ValueType)
  type?: 'percentage' | 'number';
}

async function validateSettings(settings: any) {
  try {
    const settingsInstance = plainToClass(SettingsValidation, settings);
    
    await validateOrReject(settingsInstance);
  } catch (errors) {
    const errorMessages = errors
      .map((error: any) => Object.values(error.constraints))
      .flat()
      .join(', ');
      
    throw new FieldMetadataException(
      `Value for settings is invalid: ${errorMessages}`,
      FieldMetadataExceptionCode.INVALID_FIELD_INPUT
    );
  }
}
```

File: packages/twenty-front/src/modules/object-record/record-field/meta-types/hooks/useNumberField.ts [hunk 3]
```diff
@@ -32,14 +33,27 @@ export const useNumberField = () => {
 
   const persistField = usePersistField();
 
-  const persistNumberField = (newValue: string) => {
-    if (!canBeCastAsNumberOrNull(newValue)) {
-      return;
-    }
+  const persistNumberField = (newValue: string, isPercentage: boolean) => {
+    if (isPercentage) {
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-field/meta-types/hooks/useNumberField.ts:37]: let's avoid nested if and do early returns :) this will easy reading the code

File: packages/twenty-front/src/modules/object-record/record-field/meta-types/input/components/NumberFieldInput.tsx [hunk 4]
```diff
@@ -28,27 +28,30 @@ export const NumberFieldInput = ({
     persistNumberField,
   } = useNumberField();
 
+  const isPercentage =
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-field/meta-types/input/components/NumberFieldInput.tsx:31]: no need to pass isPercentage from there, it can be directly read from persistNumberField

File: packages/twenty-front/src/modules/object-record/record-field/types/FieldMetadata.ts [hunk 5]
```diff
@@ -46,74 +49,90 @@ export type FieldNumberMetadata = {
...
+    decimals?: number;
+    type?: 'percentage' | 'number';
+  };
 };
 
 export type FieldLinkMetadata = {
   objectMetadataNameSingular?: string;
   placeHolder: string;
   fieldName: string;
+  settings?: object;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-field/types/FieldMetadata.ts:62]: can we make it {} instead of object?

Comment (charlesBochet) [reply packages/twenty-front/src/modules/object-record/record-field/types/FieldMetadata.ts:62]: if it breaks the typing, we can keep it as it is for now

File: packages/twenty-front/src/modules/settings/data-model/object-details/components/SettingsObjectFieldItemTableRow.tsx [hunk 6]
```diff
@@ -90,7 +90,7 @@ export const SettingsObjectFieldItemTableRow = ({
       () => getRelationMetadata({ fieldMetadataItem }),
       [fieldMetadataItem, getRelationMetadata],
     ) ?? {};
-
+  console.log('fieldMetadataItem:', fieldMetadataItem);
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/settings/data-model/object-details/components/SettingsObjectFieldItemTableRow.tsx:93]: console.log!

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata-validation.service.ts (hunk 2 above)
Comment (charlesBochet) [reply packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata-validation.service.ts:36]: +1

**Review by charlesBochet — COMMENTED** [review]
Good job! Left a few comments and we can merge!

**Comment by guillim** [comment]
feedback from Charles : disable percentage option on standard object !

**Review by charlesBochet — APPROVED** [review]
LGTM, bravo!
